### ISSUE TITLE
Use signature of `__init_subclass__` of superclasses to inform keyword-argument completion suggestions inside class definitions

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ty#1777]
In this snippet, the class definition `Bar` will fail because `Foo.__init_subclass__` states that subclasses must provide the `special_keyword` keyword argument when you inherit from `Foo`:

```py
class Foo:
    def __init_subclass__(cls, special_keyword):
        pass

class Bar(Foo): ...
```

The definition of `Bar` must instead be:

```py
class Bar(Foo, special_keyword=42): ...
```

or similar. We could look at the `__init_subclass__` signatures of classes already in the class definition to inform autocomplete suggestions for keyword arguments in this context.

**Comment by carljm** [comment]
It seems like the type checker should also error on the definition of `Bar` that doesn't provide `special_keyword`? I guess that should be a separate issue? (Maybe you already filed one and I didn't see it yet.)

**Comment by AlexWaygood** [comment]
yeah I filed that one a few weeks/months ago ;)

**Comment by AlexWaygood** [comment]
see https://github.com/astral-sh/ty/issues/1541

**Comment by dangotbanned** [comment]
Sorely missing this in `pyright`/`pylance`, would be great to see better support in `ty`

- https://github.com/microsoft/pylance-release/discussions/2781

(I assume these are related, but ignore me if not)

The lacking `__init_subclass__` support in `pylance` *also* means symbols that are used in the defining context don't show up in `Find All References` either
